pkg/controller/plan/controller.go

```diff
@@ -204,8 +210,19 @@ func (r Reconciler) Reconcile(request reconcile.Request) (result reconcile.Resul
 		r.Log.V(2).Info("Conditions.", "all", plan.Status.Conditions)
 	}()
 
-	// Don't reconcile if the plan is archived.
-	if plan.Spec.Archived && plan.Status.HasCondition(Archived) {
+	// Set archive finalizer to ensure resources are
+	// cleaned up if the plan is deleted without
+	// having been manually archived.
+	if plan.DeletionTimestamp.IsZero() && !plan.HasFinalizer(ArchiveFinalizer) {
+		plan.AddFinalizer(ArchiveFinalizer)
+		err = r.Update(context.TODO(), plan)
```

Comment:
@jortel do you think there would be any issue with changing the Plan status subresource `Update()` call at the end of `Reconcile()` to just update the whole plan? That would allow me to leave out the `Update()` call here.